### PR TITLE
[SYCL][Graph] Update design doc for copy optimization and add test 

### PR DESCRIPTION
**Pull request by mfrancepillois**
- Update UR tag to include L0 command-buffer copy engine optimization
- Add test which mixes copy and kernel commands
- Update design doc to detail copy engine optimization

Co-authored By:

* Ben Tracy <ben.tracy@codeplay.com>
* Ewan Crawford <ewan@codeplay.com>

### CODE REVIEW

**Review by EwanC — APPROVED** [review]
UR tag will need updated once PR is merged, but graph changes LGTM

File: sycl/plugins/unified_runtime/CMakeLists.txt [hunk 1]
```diff
@@ -109,14 +109,8 @@ if(SYCL_PI_UR_USE_FETCH_CONTENT)
...
-    ${UNIFIED_RUNTIME_REPO}
-    # commit 8788bd13cceb3f8e6338538b624652e6249a4543
-    # Merge: 78d02039 3f502d8f
-    # Author: Kenneth Benzie (Benie) <k.benzie@codeplay.com>
-    # Date:   Wed Jun 12 13:13:52 2024 +0100
-    #     Merge pull request #1697 from againull/review/againull/l0_loader
-    #     [L0] Add flexibility to change level zero repo
-    8788bd13cceb3f8e6338538b624652e6249a4543
+    "https://github.com/bensuo/unified-runtime.git"
+    "cmd-buf-copy-queue"
```
Comment (kbenzie) [inline sycl/plugins/unified_runtime/CMakeLists.txt:113]:
```suggestion
    ${UNIFIED_RUNTIME_REPO}
    # commit b8a1a3f232198bf2c3d8edd2bbc909bb2a9be555
    # Merge: 0cd127ad 30f8ac50
    # Author: Kenneth Benzie (Benie) <k.benzie@codeplay.com>
    # Date:   Fri Jun 14 14:26:17 2024 +0100
    #     Merge pull request #1738 from Bensuo/cmd-buf-copy-queue
    #     [CMDBUF][L0] Use copy engine to optimize cmd-buffer usage
    b8a1a3f232198bf2c3d8edd2bbc909bb2a9be555
```

**Review by kbenzie — COMMENTED** [review]
https://github.com/oneapi-src/unified-runtime/pull/1738 has been merged, please pull in the latest `sycl` branch changes and update the UR repo/tag as suggested.

**Review by kbenzie — APPROVED** [review]

**Comment by kbenzie** [comment]
@intel/llvm-gatekeepers please merge

**Pull request merged**